src/test/scala/com/wavesplatform/TransactionGen.scala

```diff
@@ -159,12 +159,13 @@ trait TransactionGen {
     feeScale2: Short <- feeScaleGen
   } yield (lease, LeaseCancelTransaction.create(sender, lease.id, cancelFee, feeScale2, timestamp + 1).right.get)
 
-  def leaseAndCancelGeneratorP(leaseSender: PrivateKeyAccount, recipient: AddressOrAlias, unleaseSender: PrivateKeyAccount): Gen[(LeaseTransaction, LeaseCancelTransaction)] = for {
-    (_, amount, fee, feeScale, timestamp, _) <- leaseParamGen
+  def leaseAndCancelGeneratorP(leaseSender: PrivateKeyAccount, recipient: AddressOrAlias, ts: Long = 0): Gen[(LeaseTransaction, LeaseCancelTransaction)] = for {
```

Comment:
why remove unleaseSender here? is there any cases we need to test that `leaseSender` \neq to `unleaseSender`?

Reply:
`unleaseSender` must be equal to `leaseSender`
other account cannot cancel the the lease tx sent by lease sender

Reply:
the from the design view "yes", but for test i am not sure wether we need such invalid case.

Reply:
If they are different, it will cause an error when written to state.
I've checked all the code related to this,  that `leaseSender and `unleaseSender`  are all equal.

Reply:
Oh, I got it. You mean that we may need to test if someone could be able to cancel other's lease tx.

Reply:
yes, i totally understood ur point. but invalid case in unit test is important too. it seems that we didn't add the case to the unit test. which means that line
https://github.com/virtualeconomy/v-systems/blob/cd5d3703ed49421551717aaa7781f97f059859d1/src/main/scala/com/wavesplatform/state2/diffs/LeaseTransactionsDiff.scala#L57
has not been tested well.

Reply:
I think the `leaseAndCancelWithOtherSenderGen` is enough.

Reply:
I'm adding it to lease unit test

Reply:
i see, great. `leaseAndCancelWithOtherSenderGen` looks redundant~ 

overall, just choose one of them~ ;)

Reply:
Actually....... https://github.com/virtualeconomy/v-systems/blob/cd5d3703ed49421551717aaa7781f97f059859d1/src/test/scala/com/wavesplatform/state2/diffs/LeaseTransactionsDiffTest.scala#L150

Reply:
sure. just keep it >.<